core/ir/aarch64/codec.txt

```diff
@@ -294,1336 +294,1347 @@ x----------xxxxx----------------  wx16       # W/X register (or WZR/XZR)
+11001000000xxxxx0^^^^^xxxxxxxxxx  n   467        stxr       mem0 w16 : x0
+00001000000xxxxx0^^^^^xxxxxxxxxx  n   468       stxrb       mem0 w16 : w0
+01001000000xxxxx0^^^^^xxxxxxxxxx  n   469       stxrh       mem0 w16 : w0
+x10100010xxxxxxxxxxxxxxxxxxxxxxx  n   470         sub          wx0sp : wx5sp imm12 lsl imm12sh
+x1001011xx0xxxxxxxxxxxxxxxxxxxxx  n   470         sub            wx0 : wx5 wx16 shift3 imm6
+x1001011001xxxxxxxxxxxxxxxxxxxxx  n   470         sub          wx0sp : wx5sp wx16 ext extam
+0x101110xx1xxxxx100001xxxxxxxxxx  n   470         sub            dq0 : dq5 dq16 bhsd_sz
+01111110111xxxxx100001xxxxxxxxxx  n   470         sub             d0 : d5 d16
+00001110xx1xxxxx011000xxxxxxxxxx  n   471       subhn             d0 : q5 q16 bhs_sz
+01001110xx1xxxxx011000xxxxxxxxxx  n   472      subhn2             q0 : q5 q16 bhs_sz
+x11100010xxxxxxxxxxxxxxxxxxxxxxx  w   473        subs            wx0 : wx5sp imm12 lsl imm12sh
+x1101011xx0xxxxxxxxxxxxxxxxxxxxx  w   473        subs            wx0 : wx5 wx16 shift3 imm6
+x1101011001xxxxxxxxxxxxxxxxxxxxx  w   473        subs            wx0 : wx5sp wx16 ext extam
+0101111000100000001110xxxxxxxxxx  n   474      suqadd             b0 : b5
+0101111001100000001110xxxxxxxxxx  n   474      suqadd             h0 : h5
+0101111010100000001110xxxxxxxxxx  n   474      suqadd             s0 : s5
+0101111011100000001110xxxxxxxxxx  n   474      suqadd             d0 : d5
+0x001110xx100000001110xxxxxxxxxx  n   474      suqadd            dq0 : dq5 bhsd_sz
+11010100000xxxxxxxxxxxxxxxx00001  n   475         svc                : imm16
+10111000001xxxxx100000xxxxxxxxxx  n   476         swp        w0 mem0 : w16 mem0
+11111000001xxxxx100000xxxxxxxxxx  n   476         swp        x0 mem0 : x16 mem0
+10111000101xxxxx100000xxxxxxxxxx  n   477        swpa        w0 mem0 : w16 mem0
+11111000101xxxxx100000xxxxxxxxxx  n   477        swpa        x0 mem0 : x16 mem0
+00111000101xxxxx100000xxxxxxxxxx  n   478       swpab        w0 mem0 : w16 mem0
+01111000101xxxxx100000xxxxxxxxxx  n   479       swpah        w0 mem0 : w16 mem0
+10111000111xxxxx100000xxxxxxxxxx  n   480       swpal        w0 mem0 : w16 mem0
+11111000111xxxxx100000xxxxxxxxxx  n   480       swpal        x0 mem0 : x16 mem0
+00111000111xxxxx100000xxxxxxxxxx  n   481      swpalb        w0 mem0 : w16 mem0
+01111000111xxxxx100000xxxxxxxxxx  n   482      swpalh        w0 mem0 : w16 mem0
+00111000001xxxxx100000xxxxxxxxxx  n   483        swpb        w0 mem0 : w16 mem0
+01111000001xxxxx100000xxxxxxxxxx  n   484        swph        w0 mem0 : w16 mem0
+10111000011xxxxx100000xxxxxxxxxx  n   485        swpl        w0 mem0 : w16 mem0
+11111000011xxxxx100000xxxxxxxxxx  n   485        swpl        x0 mem0 : x16 mem0
+00111000011xxxxx100000xxxxxxxxxx  n   486       swplb        w0 mem0 : w16 mem0
+01111000011xxxxx100000xxxxxxxxxx  n   487       swplh        w0 mem0 : w16 mem0
+1101010100001xxxxxxxxxxxxxxxxxxx  n   488         sys                : sysops memx0
```

Comment:
So we continue to have this entry. Is that for all variants of `sys` other than `dc` or `ic` (which are handled by a different entry above)?

I may have missed previous decoder changes that allow such overlapping entries in the codec (the bits of `sys` and `dc` match except the `x`s). What's the logic now? When there are multiple matching entries, is the first matching entry used?

Reply:
>So we continue to have this entry. Is that for all variants of sys other than dc or ic (which are handled by a different entry above)?

Yes, all variants of SYS other than DC or IC.

> I may have missed previous decoder changes that allow such overlapping entries in the codec (the bits of sys and dc match except the xs). What's the logic now? When there are multiple matching entries, is the first matching entry used?

The codec can now resolve conflicts to the precise instruction from the encoding regardless of its postion in `codec.txt`.

The reason the overlaps were problematic was because resolution was based on the fact that operand overlaps using `x` allowed for ANY combination of bits. But similar encodings are designed such that operands can only have a limited number of bit patterns. The `+` operand mask was introduced in #5092 to resolve such conflicts.

Reply:
Ack.